tests/Maester/Entra/Test-MtMdiHealthIssues.Tests.ps1

```diff
@@ -0,0 +1,71 @@
+        Add-MtTestResultDetail -SkippedBecause NoResults
+        return $null
+    } else {
+        $MdiHealthIssues = New-Object System.Collections.ArrayList
+
+        # Add domainNames and sensorDNSNames as string properties to identify unique health issues
+        $MdiAllHealthIssues | ForEach-Object {
+            $_ | Add-Member -NotePropertyName 'domainNamesString' -NotePropertyValue ($_.domainNames -join ',') -Force
+            $_ | Add-Member -NotePropertyName 'sensorDNSNamesString' -NotePropertyValue ($_.sensorDNSNames -join ',') -Force
+        }
+
+        # Get unique health issues (duplicated entries will be created when status of an issue has been changed)
+        $MdiAllHealthIssues | Group-Object -Property displayName, domainNamesString, sensorDNSNamesString | ForEach-Object {
+            $UniqueHealthIssue = $_.Group | Sort-Object -Property createdDateTime | Select-Object -First 1
+
+            # Add the displayName to the health issue to avoid confusion of same health issue name
+            if ($UniqueHealthIssue.displayName -eq "Sensor stopped communicating") {
+                $UniqueHealthIssue.displayName = $UniqueHealthIssue.displayName + " - " + $UniqueHealthIssue.sensorDNSNames
+            }
+            $MdiHealthIssues.Add($UniqueHealthIssue) | Out-Null
+        }
+
+        $MdiHealthActiveIssues = $MdiHealthIssues | Where-Object { $_.status -ne "closed" }
+    }
+}
+
+Describe "Defender for Identity health issues" -Tag "Maester", "Entra", "Security", "All", "MDI" -ForEach $MdiHealthActiveIssues {
+    It "MT.1058: MDI Health Issues - <displayName>. See https://maester.dev/docs/tests/MT.1058" -Tag "MT.1058", $displayName {
+
+        $issueUrl = "https://security.microsoft.com/identities/health-issues"
+        $recommendationLinkMd = "`n`n➡️ Open [Health issue - $displayName]($issueUrl) in the Microsoft Defender portal."
+        $testTitle = "MT.1058.$($id): MDI Health Issues - $displayName"
+
+        if ( $status -match "dismissed" -or $status -match "suppressed" ) {
+            Add-MtTestResultDetail -TestTitle $testTitle -Description $description -SkippedBecause Custom -SkippedCustomReason "This health issue has been **Suppressed** by an administrator.`n`nIf this issue is valid for your MDI instance you can change it's state from **suppressed** to **Re-open**. $recommendationLinkMd"
+            return $null
+        }
+
+
+        #region Add detailed test description
+        $actionSteps = $recommendations | ForEach-Object {
+            "- " + $_
+        }
+        $actionSteps = $actionSteps -join "`n`n"
+
+        #
+        $affectedItems = $additionalInformation | ForEach-Object {
+            "- " + $_
+        }
+        $affectedItems = $affectedItems -join "`n`n"
+
+        $ResultMarkdown = $description + "`n`n" + $affectedItems + "`n`n#### Remediation actions:`n`n" + $actionSteps  + "`n`n#### Issue updated:`n`n" + $lastModifiedDateTime + "`n`n#### Issue created:`n`n" + $createdDateTime
```

Comment:
I would recommend line breaks after some of the `+` operators to make this easier to read & review.